### PR TITLE
V3.0.0

### PR DESCRIPTION
**Pull request by GianlucaGuarini**
The new riot documentation is almost ready I just need to add the migration guide from riot2 to riot3.
Let me know what you think please

### CODE REVIEW

File: release-notes.md [hunk 1]
```diff
@@ -6,4 +6,71 @@ id: release-notes
...
+
+It has been a long road but in the end we made it and riot@3.0.0 is finally out and it represents a
+step forward compared to riot 2 for stability, performance and flexibility.
+
+List of improvements and breaking changes
+
+__riot__:
+
+  - _Change:_ build riot using rollup+babel instead of smash using only es6 modules syntax
+  - _Change:_ clean up the tests splitting them into several files, now our unit test was rewritten using chai.js and es6 syntax
```
Comment (crisward) [inline release-notes.md]: clean up the tests splitting them into several files, now our unit test 'are' rewritten using chai.js and es6 syntax

Comment (GianlucaGuarini) [inline release-notes.md]: @crisward yeah you are right I need to rephrase this thanks

**Comment by GianlucaGuarini** [comment]
I am ready here. We could release riot@3.0.0. 

@riot/core-maintainers please check if everything is fine for you as well and we could make the new release

✌️ 

p.s.
We will update the riot-examples afterwards, we can add a message in the readme file saying that all the examples will be updated soon

**Comment by MartinMuzatko** [comment]
I will update the [Riot Cheatsheet](http://martinmuzatko.github.io/riot-cheatsheet/) ASAP.  Thanks for the new documentation!

File: guide/compiler.md [hunk 2]
```diff
@@ -280,37 +276,17 @@ A sample tag written in ES6:
...
-
-Babel 6 introduced many breaking changes so if you want to use it you should configure your environment first:
+Before using the es6 compiler you should properly configure your project following the steps below:
 
  1. install our [babel-preset-es2015-riot](https://github.com/riot/babel-preset-es2015-riot)<br /> `npm install babel-preset-es2015-riot --save-dev`
  2. install `babel-core` as well <br /> `npm install babel-core -g`
  3. create a `.babelrc` file containing the preset id<br /> `{ "presets": ["es2015-riot"] }`
 
-Once your have configured your environment you can use:
+Once your have configured your environment you can compile your tags:
```
Comment (MartinMuzatko) [inline guide/compiler.md]: Once you have configured*

File: guide/index.md [hunk 3]
```diff
@@ -74,25 +74,7 @@ A Riot tag is a combination of layout (HTML) and logic (JavaScript). Here are th
...
-<my-tag></my-tag>
-
-  <!-- this fails, because of indentation -->
-  <my-tag>
-
-  </my-tag>
-```
-
-Inline tag definitions(in document body) must be properly indented, with all custom tags equally indented at the lowest indent level, mixing of tabs and spaces is discouraged.
+Inline tag definitions(in document body) should be properly indented, with all custom tags equally indented at the lowest indent level, mixing of tabs and spaces is discouraged.
```
Comment (MartinMuzatko) [inline guide/index.md]: Put in a space before (in document body) - we use a space almost anywhere else.

File: guide/index.md [hunk 4]
```diff
@@ -220,30 +202,12 @@ A document can contain multiple instances of the same tag.
...
-  this.on('update', function(){
-    var test2 = document.getElementById('findMe')
-    console.log('test2', test2) // Succeeds
-  })
-  </script>
-</example-tag>
-```
-
-You probably don't want to run whatever you're attempting to retrieve on every update. Instead you'll most likely want to run on the `mount` event.
+If you need to access the DOM inside Riot, you'll want to take a look at the [Tag Lifecycle](#tag-lifecycle) and notice that the DOM elements aren't instantiated until the `mount` event first fires, meaning any attempt to select an element before then will fail.
```
Comment (MartinMuzatko) [inline guide/index.md]: We should split this sentence. 

> If you need to access the DOM inside Riot, you'll want to take a look at the [Tag Lifecycle](#tag-lifecycle). Notice that the DOM elements aren't instantiated until the `mount` event first fires, meaning any attempt to select an element before then will fail.

**Comment by GianlucaGuarini** [comment]
@MartinMuzatko thanks for your feedback. I will fix and do the release tonight

**Comment by GianlucaGuarini** [comment]
@MartinMuzatko I have fixed the issues and I was thinking to integrate the Riot Cheatsheet directly on our site. Would you mind making a pull request when you will have time please?

**Comment by GianlucaGuarini** [comment]
Riot@3.0.0 was finally published! Thanks to anyone

**Comment by MartinMuzatko** [comment]
@GianlucaGuarini I'd love to! Please let me know further implementation details in issue #162.

**Pull request merged**